pkg/karmadactl/util/check.go

```diff
@@ -0,0 +1,84 @@
+package util
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog/v2"
+)
+
+// WaitForStatefulsetRollout wait statefulset rollout
+func WaitForStatefulsetRollout(c kubernetes.Interface, sts *appsv1.StatefulSet, timeout int) error {
+	var lastErr error
+	if err := wait.Poll(time.Second, time.Duration(timeout)*time.Second, func() (bool, error) {
+		s, err := c.AppsV1().StatefulSets(sts.GetNamespace()).Get(context.TODO(), sts.GetName(), metav1.GetOptions{})
+		if err != nil {
+			lastErr = err
+			klog.Warningf("WaitForStatefulsetRollout: failed to get StatefulSet %s/%s", sts.GetNamespace(), sts.GetName())
+			return false, nil
+		}
+		if s.Generation > s.Status.ObservedGeneration {
+			lastErr = fmt.Errorf("expected generation %d, observed generation: %d",
+				s.Generation, s.Status.ObservedGeneration)
+			return false, nil
+		}
+		if s.Status.UpdatedReplicas != s.Status.Replicas {
+			lastErr = fmt.Errorf("expected %d replicas, got %d updated replicas",
+				s.Status.Replicas, s.Status.UpdatedReplicas)
+			return false, nil
+		}
+		if s.Status.ReadyReplicas != s.Status.Replicas {
+			lastErr = fmt.Errorf("expected %d replicas, got %d ready replicas",
+				s.Status.Replicas, s.Status.ReadyReplicas)
+			return false, nil
+		}
+		return true, nil
+	}); err != nil {
+		if err == wait.ErrWaitTimeout && lastErr != nil {
+			err = lastErr
+		}
+		return fmt.Errorf("failed to wait for StatefulsetRollout of %s/%s ue to %w", sts.GetNamespace(), sts.GetName(), err)
+	}
+	return nil
+}
+
+// WaitForDeploymentRollout wait deployment rollout
+func WaitForDeploymentRollout(c kubernetes.Interface, dep *appsv1.Deployment, timeout int) error {
+	var lastErr error
+	if err := wait.Poll(time.Second, time.Duration(timeout)*time.Second, func() (bool, error) {
```

Comment:
Use `wait.PollImmediate` is a better choice, make it faster.

Reply:
thanks，i fixed it